src/states/elder/mod.rs

```diff
@@ -1507,17 +1507,37 @@ impl Elder {
             }
         }
 
-        let target_pub_ids = self.get_targets(signed_msg.routing_message(), hop)?;
+        // we will be retrying until either a target succeeds, or all fail for reasons other than
+        // not being connected; if a target was not connected, it will get dropped in
+        // `send_signed_message_to_peer` and won't be a potential target the next time
+        loop {
+            let target_pub_ids = self.get_targets(signed_msg.routing_message(), hop)?;
 
-        debug!(
-            "{}: Sending message {:?} via targets {:?}",
-            self,
-            signed_msg.routing_message(),
-            target_pub_ids
-        );
+            debug!(
+                "{}: Sending message {:?} via targets {:?}",
+                self,
+                signed_msg.routing_message(),
+                target_pub_ids
+            );
+
+            let send_results: Vec<_> = target_pub_ids
+                .into_iter()
+                .map(|target| self.send_signed_message_to_peer(signed_msg.clone(), &target))
+                .collect();
 
-        for target_pub_id in target_pub_ids {
-            self.send_signed_message_to_peer(signed_msg.clone(), &target_pub_id)?;
+            if send_results.iter().any(|result| result.is_ok()) {
+                break;
+            } else if send_results.iter().all(|result| match result {
+                Err(RoutingError::Interface(InterfaceError::NotConnected)) => false,
```

Comment:
Is this a match statement instead of a `!=` because `RoutingError` isn't `PartialEq`? Shouldn't it be?

Reply:
That.s the reason, yes :+1: Maybe it could be, I just went with what was there, but it would probably be cleaner with `PartialEq`.